# Experiment Validation

This notebook evaluates the experimental group allocation and balance of pre-treatment customer characteristics in the Hillstrom email marketing experiment.

The experiment contains 64,000 customers assigned to three groups: Men's Email, Women's Email, and No Email.

Two validation checks are performed:

- **Sample Ratio Mismatch (SRM):** Assess whether observed group sizes deviate from the assumed 1:1:1 allocation.
- **Covariate Balance:** Compare pre-treatment customer characteristics across experimental groups.

Data quality checks and detailed SQL calculations are documented in `sql/02_data_validation.sql` and `sql/03_experiment_validation.sql`.

In [1]:
from scipy.stats import chisquare

# Observed counts: Men's Email, Control, Women's Email
observed = [21307, 21306, 21387]
expected = [sum(observed) / 3] * 3

result = chisquare(f_obs=observed, f_exp=expected)

print(f"Chi-square statistic: {result.statistic:.6f}")
print(f"P-value: {result.pvalue:.6f}")

Chi-square statistic: 0.202531
P-value: 0.903693



## 1. Sample Ratio Mismatch (SRM)

The chi-square goodness-of-fit test yielded a statistic of 0.202531 (df = 2) and a p-value of 0.903693.

At the 0.05 significance level, we fail to reject the null hypothesis. No statistically significant evidence of a sample ratio mismatch was detected under the assumed 1:1:1 allocation.

## 2. Covariate Balance

Standardized mean differences (SMDs) were calculated for five pre-treatment customer characteristics.

| Covariate | Men's Email vs Control | Women's Email vs Control |
|---|---:|---:|
| History | 0.0076 | 0.0065 |
| Recency | 0.0068 | 0.0052 |
| Newbie | -0.0009 | 0.0026 |
| Men's merchandise history | -0.0046 | -0.0086 |
| Women's merchandise history | 0.0076 | 0.0049 |

All absolute SMDs were below 0.01, indicating negligible standardized differences in the evaluated characteristics.

## 3. Categorical Covariate Balance

Purchase channel and geographic category distributions were also compared across experimental groups.

| Covariate | Largest Difference vs Control | Category |
|---|---:|---|
| Purchase channel | 0.55 pp | Web (Men's Email) |
| Geographic category | 0.59 pp | Surburban (Men's Email) |

No substantial descriptive imbalance was observed in either categorical variable.

## 4. Conclusion

The SRM test and baseline covariate comparisons revealed no obvious allocation imbalance in the evaluated characteristics.

These findings support proceeding with treatment effect estimation, although observed balance does not guarantee perfect randomization or rule out unobserved imbalances.

Categorical balance assessments were descriptive rather than formal statistical tests.
